src/convertToObject.js

```diff
@@ -1,12 +1,18 @@
 'use strict';
 
-/**
- * @param {string} sourceString
- *
- * @return {object}
- */
 function convertToObject(sourceString) {
-  // write your code here
+  const array = sourceString
+    .split(';')
+    .map((str) => str.trim())
+    .filter((str) => str !== '')
+    .map((str) => str.split(/\s*:\s*/));
```

Comment:
The regular expression `/\s*:\s*/` is used correctly to split each key-value pair by the colon, allowing for optional spaces around the colon.